frontend/src/Components/RoadDetails/ImageGallery.tsx

```diff
@@ -70,28 +70,52 @@ const images: Image[] = [
 
 const ImageGallery: React.FC = () => {
   const [scrollPosition, setScrollPosition] = useState(0);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [showArrows, setShowArrows] = useState(true);
 
   const openImageInNewTab = (imageUrl: string) => {
     window.open(imageUrl, '_blank');
   };
 
   const handleScroll = (scrollOffset: number) => {
-    const newScrollPosition = scrollPosition + scrollOffset;
+    if (!containerRef.current) return;
 
-    // Calculate the maximum scroll limit based on the total image width
-    const maxScrollLimit = (images.length - 11.4) * 124; // 120 is the width of each image
+    const containerWidth = containerRef.current.clientWidth;
+    const totalImageWidth = images.length * 124; // 120 for the image width and 4 for margin
 
-    // Ensure the scroll stays within bounds
-    const boundedScrollPosition = Math.max(
-      0,
-      Math.min(newScrollPosition, maxScrollLimit),
-    );
+    const centerOffset = (containerWidth - totalImageWidth) / 2;
 
-    setScrollPosition(boundedScrollPosition);
+    let newScrollPosition = scrollPosition + scrollOffset;
+
+    // If scrolling to the left, don't go past half the container's width
+    if (scrollOffset < 0) {
+      newScrollPosition = Math.max(newScrollPosition, centerOffset);
+    }
+    // If scrolling to the right, don't go past the width of the images minus half the container's width
+    else {
+      newScrollPosition = Math.min(
+        newScrollPosition,
+        totalImageWidth - containerWidth + centerOffset,
+      );
+    }
+
+    setScrollPosition(newScrollPosition);
   };
 
+  useEffect(() => {
```

Comment:
Probably need to be refreshed is the size of the browser is changed (otherwise the array aren't shown when client resize his/her browser) .

Maybe something like that https://stackoverflow.com/questions/44480053/how-to-detect-if-screen-size-has-changed-to-mobile-in-react ?